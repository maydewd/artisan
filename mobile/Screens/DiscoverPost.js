/**
 * screen for viewing discover post information
 */

import React, {Component} from 'react';
import {
  AppRegistry,
  StyleSheet,
  Text,
  View,
  Image,
  ScrollView,
  AsyncStorage
} from 'react-native';
var NavigationBar = require('react-native-navbar');
import Button from 'react-native-button'
import {usableWithTop} from '../helpers/dimension'

styles = require('../Styles/Layouts');


class DiscoverPost extends Component {
  constructor(props) {
    super(props);
    this.state = {
      description: this.props.item.description,
      price: this.props.item.price,
      number: 0,  // TODO
      imagePath: this.props.item.imagePath,
      location: this.props.item.location,
      type: this.props.item.type
    };
  }

  render() {
    var titleConfig = {
      title: 'Discover Post',
    };
    const leftButtonConfig = {
      title: 'Back',
      handler: () => {this.pop()}
    };
    // TODO: put uri somewhere else
    return (
      <View>
        <NavigationBar
          style={styles.navBar}
          title={titleConfig}
          leftButton={leftButtonConfig}
        />
        <ScrollView style = {{height: usableWithTop()}}>
          <View
            style = {styles.discoverPostContainer}>
            <Image style={styles.discoverPostImage} source={{uri: "http://colab-sbx-137.oit.duke.edu:3000/" + this.state.imagePath}} />
          </View>
          <Text
            style={{height: 130, padding: 5, borderTopWidth: 1, borderBottomWidth: 1, borderColor: 'gray'}}
            // multiline = {true}
            numberOfLines = {3}>
            {this.state.description}
          </Text>
          <Text
            style={{height: 40, padding: 5, borderBottomWidth: 1, borderColor: 'gray'}}>
            {this.state.location}
          </Text>
          <Text
            style={{height: 40, padding: 5, borderBottomWidth: 1, borderColor: 'gray'}}>
            {this.state.price}
          </Text>
          <Text
            style={{height: 40}}>
            {this.state.type}
          </Text>
        </ScrollView>
      </View>
    );
  }

  pop() {
    this.props.navigator.pop()
  }
}

module.exports = DiscoverPost
