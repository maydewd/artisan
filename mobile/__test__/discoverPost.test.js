import 'react-native';
import React from 'react';
import DiscoverPost from '../Screens/DiscoverPost.js';

import renderer from 'react-test-renderer';

it('DiscoverPost renders correctly', () => {
  var item = {
    description: "This is a description",
    price: 10,
    imagePath: "www.path.com",
    location: null,
    type: "Ceramic"
  }
  const tree = renderer.create(
    <DiscoverPost item = {item}/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
