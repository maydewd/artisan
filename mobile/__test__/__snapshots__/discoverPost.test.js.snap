exports[`test DiscoverPost renders correctly 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "white",
        },
        null,
      ]
    }>
    <View
      style={
        Array [
          Object {
            "height": 20,
          },
          null,
        ]
      } />
    <View
      style={
        Array [
          Object {
            "alignItems": "stretch",
            "flexDirection": "row",
            "height": 44,
            "justifyContent": "space-between",
          },
          Object {
            "backgroundColor": "#cce5ff",
            "height": 45,
          },
        ]
      }>
      <View
        style={
          Object {
            "alignItems": "center",
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#333",
                "fontSize": 17,
                "fontWeight": "500",
                "letterSpacing": 0.5,
              },
              null,
            ]
          }>
          Discover Post
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "stretch",
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
              "opacity": 1,
            }
          }
          testID={undefined}>
          <View
            style={
              Array [
                undefined,
                Object {
                  "marginLeft": 8,
                },
              ]
            }>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "fontSize": 17,
                    "letterSpacing": 0.5,
                  },
                  Object {
                    "color": "#0076FF",
                  },
                ]
              }>
              Back
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "stretch",
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
              "opacity": 1,
            }
          }
          testID={undefined}>
          <View
            style={
              Array [
                undefined,
                Object {
                  "marginRight": 8,
                },
              ]
            }>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "fontSize": 17,
                    "letterSpacing": 0.5,
                  },
                  Object {
                    "color": "#0076FF",
                  },
                ]
              }>
              
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
  <ScrollView
    style={
      Object {
        "height": 602,
      }
    }>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "space-around",
        }
      }>
      <Image
        source={
          Object {
            "uri": "http://colab-sbx-137.oit.duke.edu:3000/www.path.com",
          }
        }
        style={
          Object {
            "height": 276,
            "resizeMode": "contain",
            "width": 375,
          }
        } />
    </View>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "white",
            "flex": 1,
            "flexDirection": "row",
            "overflow": "hidden",
          },
          Object {
            "backgroundColor": "white",
            "borderColor": "gray",
            "borderTopWidth": 1,
            "flex": 3,
            "paddingTop": 10,
            "width": 375,
          },
        ]
      }>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "justifyContent": "center",
            "marginLeft": 0,
            "padding": 16,
          }
        }
        testID={undefined}>
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#24518D",
                "fontFamily": "FontAwesome",
                "fontSize": 25,
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              undefined,
            ]
          }>
          ï€­
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "left": 80,
            "opacity": 0,
            "position": "absolute",
            "top": 16,
          }
        }
        testID={undefined}>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#D2D2D2",
                "fontFamily": "Arial",
                "fontSize": 18,
                "fontWeight": "bold",
              },
              Object {
                "color": "pink",
              },
            ]
          }>
          Story
        </Text>
      </View>
      <TextInput
        easing={[Function]}
        editable={false}
        iconClass={[Function]}
        iconColor="#24518D"
        iconName="book"
        inputStyle={
          Object {
            "color": "#24518D",
          }
        }
        label="Story"
        labelStyle={
          Object {
            "color": "pink",
          }
        }
        multiline={true}
        numberOfLines={2}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        style={
          Array [
            Object {
              "color": "black",
              "flex": 1,
              "fontSize": 18,
              "paddingHorizontal": 16,
              "paddingVertical": 0,
            },
            Object {
              "color": "#24518D",
            },
          ]
        }
        underlineColorAndroid="transparent"
        value="This is a description" />
    </View>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "white",
            "flex": 1,
            "flexDirection": "row",
            "overflow": "hidden",
          },
          Object {
            "backgroundColor": "white",
            "borderColor": "gray",
            "borderTopWidth": 1,
            "flex": 1,
            "width": 375,
          },
        ]
      }>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "justifyContent": "center",
            "marginLeft": 0,
            "padding": 16,
          }
        }
        testID={undefined}>
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#24518D",
                "fontFamily": "FontAwesome",
                "fontSize": 25,
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              undefined,
            ]
          }>
          ï‚¬
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "left": 80,
            "opacity": 0,
            "position": "absolute",
            "top": 16,
          }
        }
        testID={undefined}>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#D2D2D2",
                "fontFamily": "Arial",
                "fontSize": 18,
                "fontWeight": "bold",
              },
              Object {
                "color": "pink",
              },
            ]
          }>
          Location
        </Text>
      </View>
      <TextInput
        easing={[Function]}
        editable={false}
        iconClass={[Function]}
        iconColor="#24518D"
        iconName="globe"
        inputStyle={
          Object {
            "color": "#24518D",
          }
        }
        label="Location"
        labelStyle={
          Object {
            "color": "pink",
          }
        }
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        style={
          Array [
            Object {
              "color": "black",
              "flex": 1,
              "fontSize": 18,
              "paddingHorizontal": 16,
              "paddingVertical": 0,
            },
            Object {
              "color": "#24518D",
            },
          ]
        }
        underlineColorAndroid="transparent"
        value="N/A" />
    </View>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "white",
            "flex": 1,
            "flexDirection": "row",
            "overflow": "hidden",
          },
          Object {
            "backgroundColor": "white",
            "borderColor": "gray",
            "borderTopWidth": 1,
            "flex": 1,
            "width": 375,
          },
        ]
      }>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "justifyContent": "center",
            "marginLeft": 0,
            "padding": 16,
          }
        }
        testID={undefined}>
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#24518D",
                "fontFamily": "Material Icons",
                "fontSize": 25,
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              undefined,
            ]
          }>
          î¢¡
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "left": 80,
            "opacity": 0,
            "position": "absolute",
            "top": 16,
          }
        }
        testID={undefined}>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#D2D2D2",
                "fontFamily": "Arial",
                "fontSize": 18,
                "fontWeight": "bold",
              },
              Object {
                "color": "pink",
              },
            ]
          }>
          Asking Price
        </Text>
      </View>
      <TextInput
        easing={[Function]}
        editable={false}
        iconClass={[Function]}
        iconColor="#24518D"
        iconName="payment"
        inputStyle={
          Object {
            "color": "#24518D",
          }
        }
        label="Asking Price"
        labelStyle={
          Object {
            "color": "pink",
          }
        }
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        style={
          Array [
            Object {
              "color": "black",
              "flex": 1,
              "fontSize": 18,
              "paddingHorizontal": 16,
              "paddingVertical": 0,
            },
            Object {
              "color": "#24518D",
            },
          ]
        }
        underlineColorAndroid="transparent"
        value="10" />
    </View>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "white",
            "flex": 1,
            "flexDirection": "row",
            "overflow": "hidden",
          },
          Object {
            "borderColor": "gray",
            "borderTopWidth": 1,
            "width": 375,
          },
        ]
      }>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "justifyContent": "center",
            "marginLeft": 0,
            "padding": 16,
          }
        }
        testID={undefined}>
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#24518D",
                "fontFamily": "Material Icons",
                "fontSize": 25,
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              undefined,
            ]
          }>
          î£’
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "left": 80,
            "opacity": 0,
            "position": "absolute",
            "top": 16,
          }
        }
        testID={undefined}>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#D2D2D2",
                "fontFamily": "Arial",
                "fontSize": 18,
                "fontWeight": "bold",
              },
              Object {
                "color": "pink",
              },
            ]
          }>
          Type
        </Text>
      </View>
      <TextInput
        easing={[Function]}
        editable={false}
        iconClass={[Function]}
        iconColor="#24518D"
        iconName="subject"
        inputStyle={
          Object {
            "color": "#24518D",
          }
        }
        label="Type"
        labelStyle={
          Object {
            "color": "pink",
          }
        }
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        style={
          Array [
            Object {
              "color": "black",
              "flex": 1,
              "fontSize": 18,
              "paddingHorizontal": 16,
              "paddingVertical": 0,
            },
            Object {
              "color": "#24518D",
            },
          ]
        }
        underlineColorAndroid="transparent"
        value="Ceramic" />
    </View>
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits="button"
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#24518D",
          "borderRadius": 2,
          "maxHeight": 50,
          "opacity": 1,
          "overflow": "hidden",
          "padding": 10,
        }
      }
      testID={undefined}>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontFamily": "FontAwesome",
              "fontSize": 33.12,
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            undefined,
          ]
        }>
        ïƒ 
      </Text>
    </View>
  </ScrollView>
</View>
`;
